UC NetCode 2024 Fall/Assets/App/Resource/Scripts/Player/ServerPlayerMovement.cs

```diff
@@ -0,0 +1,107 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Unity.Netcode;
+using Unity.Netcode.Components;
+using UnityEngine.EventSystems;
+
+[RequireComponent(typeof(CharacterController))]
+public class ServerPlayerMovement : NetworkBehaviour
+{
+    [SerializeField] private Animator _myAnimator;
+    [SerializeField] private NetworkAnimator _myNetAnimator;
+    [SerializeField] private float _pSpeed;
+    [SerializeField] private BulletSpawner _bulletSpawner;
+    public CharacterController _CC;
+    private MyPlayerInputActions _playerInput;
+    private static readonly int IsSprinting = Animator.StringToHash("IsSprinting");
+    private static readonly int IsWalking = Animator.StringToHash("IsWalking");
+    Vector3 _moveDirection = new Vector3(0, 0f, 0);
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        if (_myAnimator == null)
+        {
+            _myAnimator = gameObject.GetComponent<Animator>();
+        }
+
+        if (_myNetAnimator == null)
+        {
+            _myNetAnimator = gameObject.GetComponent<NetworkAnimator>();
+        }
+
+        _playerInput = new MyPlayerInputActions();
+        _playerInput.Enable();
+
+    }
+
+    // Update is called once per frame
+    void FixedUpdate()
+    {
+        if(!IsOwner) return;
+
+        // Read out player input from out new input system
+        Vector2 moveInput = _playerInput.Player.Movement.ReadValue<Vector2>();
+
+        bool isJumping = _playerInput.Player.Jumping.triggered;
+        bool isPunching = _playerInput.Player.Punching.triggered;
+        bool isSprinting = _playerInput.Player.Sprinting.triggered;
+
+        // Determine if we are a server or a player
+        if (IsServer)
+        {
+            // Move if server
+            Move(moveInput, isPunching, isSprinting, isJumping);
+        }
+        else if (IsClient && IsHost)
```

Comment:
Hello Professor NeCamp,

     I just have a quick question, if we submit no further changes to our old pull requests, will our current grade be our final grade?  I'll have some time to look at your comment and work on my pull request later today, but I'm so overwhelmed with work that if our current grade is accurate to our final grade, I'll likely take my grade as is to lessen my workload. I genuinely enjoyed your class, and I had fun doing the work, I appreciate when a Professor kind of holds your hand so you can actually learn instead of rushing to finish the work.

Have a wonderful day,
Thank you.

Alicia J Davoyan.